server/plugins/content.ts

```diff
@@ -0,0 +1,103 @@
+import { EOL } from 'os';
+import { join } from 'path';
+import { readFileSync } from 'fs';
+
+// files cache
+const files = new Map<string, string>();
+
+export default defineNitroPlugin((nitroApp) => {
+  nitroApp.hooks.hook('content:file:beforeParse', (file) => {
+    if (file.body.includes(':code-import{filePath')) {
+      const newBody = handleCodeImport(file.body);
+      file.body = newBody;
+    }
+  });
+});
+
+function handleCodeImport(body: string) {
```

Comment:
It would be nice to add a doc-comment here telling something like:

```js
// Scans the file to look for `:code-import{filePath:<path/to/file:anchor_name>}` templates, and replaces
// them with the corresponding code snippet from the `examples` directory. `anchor_name` is optional.
//
// The code snippets are also scanned for the `ANCHOR` and `ANCHOR_END` markers,
// which are used to mark that only the part of the snippet should be used.
```

And probably later we need to think about writing developer documentation on writing docs in general,
which would cover other similar extensions (not suggesting doing it in this PR).